src/AutoRest.CSharp/Common/AutoRest/Plugins/GeneratedCodeWorkspace.cs

```diff
@@ -91,6 +110,34 @@ private async Task<Document> ProcessDocument(Document document)
             return document;
         }
 
+        private bool ContainsSuppressedType(SyntaxTree syntaxTree, SemanticModel semanticModel)
```

Comment:
Why isn't this logic in MemberRemoverRewriter?

What if there are multiple types per file?

Reply:
Can we have multiple files per type? I thought that will produce a compiler error.

Reply:
Some analyzers might but I don't think we should build our logic based on this assumption.

Reply:
Ok, let me check if it will remove the whole file if there are no types.

Reply:
I think we would have to have additional logic in GeneratedCodeWorkspace to check if a file contains no types after the rewriter runs.

Reply:
Fixed